pkg/controller/elasticsearch/services/services.go

```diff
@@ -77,9 +83,30 @@ func ExternalServiceURL(es esv1.Elasticsearch) string {
 	return stringsutil.Concat(es.Spec.HTTP.Protocol(), "://", ExternalServiceName(es.Name), ".", es.Namespace, globalServiceSuffix, ":", strconv.Itoa(network.HTTPPort))
 }
 
-// NewExternalService returns the external service associated to the given cluster
+// InternalServiceURL returns the URL used to reach Elasticsearch's internally managed service
+func InternalServiceURL(es esv1.Elasticsearch) string {
+	return stringsutil.Concat(es.Spec.HTTP.Protocol(), "://", InternalServiceName(es.Name), ".", es.Namespace, globalServiceSuffix, ":", strconv.Itoa(network.HTTPPort))
+}
+
+// NewExternalService returns the external service associated to the given cluster.
 // It is used by users to perform requests against one of the cluster nodes.
 func NewExternalService(es esv1.Elasticsearch) *corev1.Service {
+	return newServiceWithName(es, ExternalServiceName(es.Name))
+}
+
+// NewInternalService returns the internal service associated to the given cluster.
+// It is used by the operator to perform requests against the elasticsearch cluster nodes,
+// and does not inherit the spec defined within the Elasticsearch custom resource,
+// to remove the possibility of the user misconfiguring access to the ES cluster.
+func NewInternalService(es esv1.Elasticsearch) *corev1.Service {
+	svc := newServiceWithName(es, InternalServiceName(es.Name))
+	svc.Spec.Selector = nil
```

Comment:
This only covers the selector. Other aspects of the service's definition are still inherited from the service template in the custom resource: e.g. type, publishNotReadyAddresses, IPFamilies etc.

Reply:
I was leaving `Type`, PublishNotReadyAddresses`, and `IPFamilies` unset as `Type` defaults to `ClusterIP`, `PublishNotReadyAddresses` defaults to false, and `IPFamilies` states:

```
//  This field is usually assigned automatically based on cluster configuration and the
// ipFamilyPolicy field
```

But I've updated all of these to be explicitly set now to remove any ambiguity.

The comments were also updated to include punctuation, and proper capitalization.  I'll be more careful in the future about these.

Reply:
Sorry my comment was maybe misleading. When I wrote "control" I did not mean to set each value but to make sure that the users cannot accidentally set them and shoot themselves in the foot.

Reply:
👍  no worries.  Explicit is better anyways.